plugins/mac-capture/mac-window-capture.m

```diff
@@ -50,28 +46,22 @@ static inline void capture_frame(struct window_capture *wc)
 	size_t width = CGImageGetWidth(img);
 	size_t height = CGImageGetHeight(img);
 
-	CGRect rect = {{0, 0}, {width, height}};
-	da_resize(wc->buffer, width * height * 4);
-	uint8_t *data = wc->buffer.array;
-
-	CGContextRef cg_context = CGBitmapContextCreate(
-		data, width, height, 8, width * 4, wc->color_space,
-		kCGBitmapByteOrder32Host | kCGImageAlphaPremultipliedFirst);
-	CGContextSetBlendMode(cg_context, kCGBlendModeCopy);
-	CGContextDrawImage(cg_context, rect, img);
-	CGContextRelease(cg_context);
+	CGDataProviderRef provider = CGImageGetDataProvider(img);
+	CFDataRef data = CGDataProviderCopyData(provider);
+	uint8_t *const bytes = (uint8_t *)CFDataGetBytePtr(data);
 	CGImageRelease(img);
 
 	struct obs_source_frame frame = {
 		.format = VIDEO_FORMAT_BGRA,
 		.width = width,
 		.height = height,
-		.data[0] = data,
+		.data[0] = bytes,
 		.linesize[0] = width * 4,
 		.timestamp = ts,
 	};
 
 	obs_source_output_video(wc->source, &frame);
+	CFRelease(data);
```

Comment:
Does `obs_source_output_video` copies the frame `data`? Else this change will introduce a use after free, if data is still used after the underlying CFData is released.

Reply:
It looks to me like it copies the data.

obs_source_output_video -> obs_source_output_video_internal -> cache_video -> copy_frame_data

Reply:
Indeed, looks like it. Then this is fine.